openapi-cli/src/test/resources/generators/client/ballerina/combination_of_apikey_and_http_oauth.bal

```diff
@@ -0,0 +1,104 @@
+import ballerina/http;
+
+# Provides API key configurations needed when communicating with a remote HTTP endpoint.
+public type ApiKeysConfig record {|
+    string apiKey;
+    string apiKey2;
+|};
+
+public isolated client class Client {
+    final http:Client clientEp;
+    final readonly & ApiKeysConfig? apiKeyConfig;
+    # Gets invoked to initialize the `connector`.
+    #
+    # + authConfig - Configurations used for Authentication
+    # + clientConfig - The configurations to be used when initializing the `connector`
+    # + serviceUrl - URL of the target service
+    # + return - An error if connector initialization failed
+    public isolated function init(AuthConfig authConfig, http:ClientConfiguration clientConfig =  {}, string serviceUrl = "https://petstore.swagger.io:443/v2") returns error? {
+        if authConfig.auth is ApiKeysConfig {
+            self.apiKeyConfig = (<ApiKeysConfig>authConfig.auth).cloneReadOnly();
+        } else {
+            clientConfig.auth = <http:OAuth2ClientCredentialsGrantConfig|http:BearerTokenConfig|http:OAuth2RefreshTokenGrantConfig>authConfig.auth;
+            self.apiKeyConfig = ();
+        }
+        http:Client httpEp = check new (serviceUrl, clientConfig);
+        self.clientEp = httpEp;
```

Comment:
We can directly say `self.clientEp = check new (serviceUrl, clientConfig);`

Reply:
Maybe if we do does it become a special case compared with the way we generate the connector? If so, fine to keep as it is.

Reply:
Keeping it as it is